R/LongitudinalQuantities.R

```diff
@@ -0,0 +1,214 @@
+.LongitudinalQuantities <- setClass(
+    "LongitudinalQuantities",
+    slots = c(
+        "quantities" = "Quantities",
+        "data" = "DataJoint",
+        "groups" = "list",
+        "time_grid" = "numeric"
+    )
+)
+
+#' @rdname LongitudinalQuantities-class
+LongitudinalQuantities <- function(
+    object,
+    groups = NULL,
+    time_grid = NULL
+) {
+    assert_that(inherits(object, "JointModelSamples"))
+    assert_that(is.null(groups) || is.character(groups))
+    data <- as.list(object@data)
+    patients <- decompose_patients(groups, names(data$pt_to_ind))
+    time_grid <- expand_time_grid(time_grid, max(data[["Tobs"]]))
+
+    gq <- generateQuantities(
+        object,
+        patients = patients$unique_values,
+        time_grid_lm = time_grid,
+        time_grid_sm = numeric(0)
+    )
+
+    quantities_raw <- extract_quantities(gq, type = "lm_identity")
+
+    quantities <- lapply(
+        patients$indexes,
+        average_samples_by_index,
+        time_index = seq_along(time_grid),
+        quantities = quantities_raw
+    )
+
+    .LongitudinalQuantities(
+        quantities = Quantities(quantities),
+        data = object@data,
+        groups = patients$groups,
+        time_grid = time_grid
+    )
+}
+
+
+
+
+#' `as.data.frame`
+#'
+#' @param x ([`LongitudinalQuantities`]) \cr Longitudinal Quantities
+#' @param ... Not used
+#' @family LongitudinalQuantities
+#' @export
+as.data.frame.LongitudinalQuantities <- function(x, ...) {
+    res <- as.data.frame(
+        x@quantities,
+        time_grid = x@time_grid,
+        groups = x@groups,
+        type = "na"
+    )
+    res["type"] <- NULL
+    return(res)
+}
+
+
+
+#' summary
+#'
+#'
+#' @description
+#' This method returns a `data.frame` of the longitudinal quantities
+#'
+#' @param object ([`LongitudinalQuantities`]) \cr Survival Quantities.
+#' @param conf.level (`numeric`) \cr confidence level of the interval.
+#' @param ... Not used.
+#'
+#' @family LongitudinalQuantities
+#' @family summary
+#' @export
+summary.LongitudinalQuantities <- function(
+    object,
+    conf.level = 0.95,
+    ...
+) {
+    res <- summary(
+        object@quantities,
+        time_grid = object@time_grid,
+        groups = object@groups,
+        type = "na",
+        conf.level = conf.level
+    )
+    res["type"] <- NULL
+    return(res)
+}
+
+
+#' Longitudinal Plot
+#'
+#' Internal plotting function to create longitudinal plots
+#' This function predominately exists to extract core logic into its own function
+#' to enable easier unit testing.
+#'
+#' @param data (`data.frame`)\cr A `data.frame` of summary statistics for longitudinal
+#' value estimates. See details.
+#' @param data_obs (`data.frame`)\cr A `data.frame` of real observed values to be
+#' overlaid for reference.  See details.
+#' @param add_ci (`logical`)\cr Should confidence intervals be added? Default = `TRUE`.
+#' @details
+#'
+#' ## `data`
+#' Should contain the following columns:
+#' - `time` - Time point
```

Comment:
this looks a bit confusing maybe just to me, how about 
```suggestion
#' - `time`: Time point
```
or so? (otherwise two dashes)